### PR TITLE
State: Add reducer and selector for WP core update status

### PR DESCRIPTION
**Pull request by tyxla**
This PR adds reducer and selector for WordPress core update status. We'll need this one after an update has completed (being successfully or not), because in `SiteIndicator` we want to show different messages upon success and error.

So, this PR hooks into the existing `sites/updates` reducer and adds a subtree to track the status. It also introduces a selector to fetch that status, storing it on a per-site basis.

To test:
* Checkout this branch.
* Verify all updates tests pass:
```
npm run test-client client/state/sites/updates/
```
* Verify the new selector tests pass:
```
npm run test-client client/state/selectors/test/get-site-core-update-status.js
```

### CODE REVIEW

File: client/state/selectors/get-site-core-update-status.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+/**
+ * Returns the WordPress core update status of a site.
+ * Returns null if the site is unknown, or WordPress core hasn't been updated yet.
+ *
+ * @param  {Object}    state   Global state tree
+ * @param  {Number}    siteId  The ID of the site we're querying
+ * @return {?Boolean}          Whether WordPress core update was successful.
+ */
+export default function getSiteCoreUpdateStatus( state, siteId ) {
+	return get( state.sites.updates.wordpressUpdateStatus, siteId, null );
```
Comment (gwwar) [inline client/state/selectors/get-site-core-update-status.js]: Maybe
```jsx
return get( state, 'sites.updates.wordpressUpdateStatus', siteId, null );
```

Comment (ockham) [inline client/state/selectors/get-site-core-update-status.js]: `sites` is guaranteed to be present tho, isn't it?

Comment (gwwar) [inline client/state/selectors/get-site-core-update-status.js]: Just an optional style thing. I prefer using get on the first object so the reader doesn't need to know what's always available.

File: client/state/selectors/get-site-core-update-status.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+/**
+ * Returns the WordPress core update status of a site.
+ * Returns null if the site is unknown, or WordPress core hasn't been updated yet.
+ *
+ * @param  {Object}    state   Global state tree
+ * @param  {Number}    siteId  The ID of the site we're querying
+ * @return {?Boolean}          Whether WordPress core update was successful.
+ */
+export default function getSiteCoreUpdateStatus( state, siteId ) {
```
Comment (gwwar) [inline client/state/selectors/get-site-core-update-status.js]: Maybe start this with `is` or `has` to give folks a hint that it should be a boolean return value. eg `isWordpressUpdateSuccesful( state, siteId )`

Comment (tyxla) [inline client/state/selectors/get-site-core-update-status.js]: Much better! Renamed.

File: client/state/selectors/get-site-core-update-status.js (hunk 1 above)
Comment (tyxla) [inline client/state/selectors/get-site-core-update-status.js]: I've addressed that suggestion, thanks!

**Comment by tyxla** [comment]
@gwwar got any other suggestions before we merge that one?

**Pull request merged**